ui/src/message_popup/message_popup_page.jsx

```diff
@@ -133,33 +181,65 @@ export default React.createClass({
       questions,
       questionsAnswered,
       shouldShowStudentCards,
-      helpType
+      shouldShowSummary,
+      helpType,
+      sessionLength
     } = this.state;
     if (!hasStarted) return this.renderInstructions();
-    if (questionsAnswered >= questions.length) return this.renderDone();
+    if (questionsAnswered >= sessionLength) return this.renderDone();
 
     const question = this.state.questions[questionsAnswered];
     return (
       <div style={styles.container}>
+        <LinearProgress mode="determinate" value={questionsAnswered} max={sessionLength} />
         <VelocityTransitionGroup enter={{animation: "slideDown"}} leave={{animation: "slideUp"}} runOnMount={true}>
           <PopupQuestion
             key={JSON.stringify(question)}
             question={question}
             shouldShowStudentCard={shouldShowStudentCards}
+            shouldShowSummary={shouldShowSummary}
             helpType={helpType}
             limitMs={90000}
             onLog={this.onLog}
-            onDone={this.onQuestionDone}/>
+            onDone={this.onQuestionDone}
+            addResponseTime={this.addResponseTime}
+            isLastQuestion={questionsAnswered+1===sessionLength ? true : false}/>
         </VelocityTransitionGroup>
+        <Snackbar
+          open={this.state.toastRevision}
+          message="Response recorded for feedback"
+          autoHideDuration={2000}
+          onRequestClose={this.removeToast}
+        />
       </div>
     );
   },
 
   renderDone() {
+    const mean = Math.round(_.mean(this.state.responseTimes)/10)/100;
+    const numUnderTime = _.filter(this.state.responseTimes, function(time){
+      return time > 90;
```

Comment:
There's a `90000` value used earlier, let's move that to a `prop` so we can have it defined in one place and then have these update when we tune it.